executor/table_reader.go

```diff
@@ -246,15 +246,14 @@ func (e *TableReaderExecutor) Next(ctx context.Context, req *chunk.Chunk) error
 
 // Close implements the Executor Close interface.
 func (e *TableReaderExecutor) Close() error {
-	if e.dummy {
-		return nil
-	}
-
 	var err error
 	if e.resultHandler != nil {
 		err = e.resultHandler.Close()
```

Comment:
Do we need to set `e.resultHandler = nil` ? It seems will not cause any problem now because `Open()` will reassign it every time. But I think it's better to reset its state to the state before Open to decouple with the implement in `Open`

Reply:
This is kind of refactor... **but if we don't have cases to cover that, it's better to keep the old code**/behaviour IMO.
I don't want any fix on fix to be introduced.